### ISSUE TITLE
printfn "%A" has bad output for discriminated unions

### ISSUE THREAD

**Issue by Pauan** [fable-compiler/Fable#810]
### Description

In `fsharpi`, using `printfn "%A"` gives a nice result for discriminated unions, but it doesn't with Fable

### Repro code

```fsharp
type DU =
    | Foo of int
    | Bar of int

printfn "%A" (Foo 1)
```

### Expected and actual results

* Expected (fsharpi)

   ```
   Foo 1
   ```

* Actual (Fable)

   ```
   {"tag":0,"data":1}
   ```

### Related information 

* fable-compiler version: 1.0.0-narumi-908
* fable-core version: 1.0.0-narumi-906
* Operating system: Debian GNU/Linux Testing (64-bit)

**Comment by alfonsogarciacaro** [comment]
Thanks for the report, I forgot to take care of printing for the new unions, I need to fix that. I also thinking that F# 4.1 gives pretty printing for unions when using the `string` operator (or `.ToString()` method), this should probably also be implemented.

**Comment by alfonsogarciacaro** [comment]
This should be fixed by updating fable-core 1.0.0-narumi-909. I also added pretty printing for unions with the `string` and `.ToString()`. Could you please check and reopen if there're still problems?

**Comment by Pauan** [comment]
@alfonsogarciacaro Yup, it's fixed, thanks! :+1:
